components/bl-mapbox-component/src/helpers.js

```diff
@@ -121,64 +121,77 @@ export const useMarkers = (markers, markersArray, setMarkersArray, mapRef, onMar
   }, [markers]);
 };
 
-export const usePolygons = (polygons, polygonsArray, setPolygonsArray, mapRef, onPolygonClick) => {
-  useEffect(() => {
-    if (polygons?.length && mapRef.current) {
-      mapRef.current.on('load', () => {
-        polygonsArray.forEach(polygon => {
-          if (mapRef.current.getSource(polygon)) {
-            mapRef.current.removeLayer(`${ polygon.id }-layer`);
-            mapRef.current.removeSource(polygon.id);
-          }
-        });
+const updatePolygonsArray = (polygons, mapRef, polygonsArray, setPolygonsArray) => {
+  if (polygons?.length && mapRef.current) {
+    mapRef.current.on('load', () => {
+      polygonsArray.forEach(polygon => {
+        if (mapRef.current.getSource(polygon)) {
+          mapRef.current.removeLayer(`${ polygon.id }-layer`);
+          mapRef.current.removeSource(polygon.id);
+        }
       });
+    });
 
-      setPolygonsArray(preparePolygons(polygons));
+    setPolygonsArray(preparePolygons(polygons));
+  }
+};
 
-      polygonsArray.forEach(polygon => {
-        mapRef.current.addSource(polygon.id, {
-          'type': 'geojson',
-          'data': {
-            'type'    : 'Feature',
-            'geometry': {
-              'type'       : 'Polygon',
-              'coordinates': [[...polygon.coordinates]],
-            },
+const addPolygons = (mapRef, polygonsArray, onPolygonClick) => {
+  mapRef.current.on('load', () => {
+    polygonsArray.forEach(polygon => {
+      mapRef.current.addSource(polygon.id, {
+        'type': 'geojson',
+        'data': {
+          'type'    : 'Feature',
+          'geometry': {
+            'type'       : 'Polygon',
+            'coordinates': [[...polygon.coordinates]],
           },
-        });
+        },
+      });
 
-        mapRef.current.addLayer({
-          'id'    : `${ polygon.id }-layer`,
-          'type'  : 'fill',
-          'source': polygon.id,
-          'layout': {},
-          'paint' : {
-            'fill-color'  : polygon.color,
-            'fill-opacity': polygon.opacity,
-          },
-        });
+      mapRef.current.addLayer({
+        'id'    : `${ polygon.id }-layer`,
+        'type'  : 'fill',
+        'source': polygon.id,
+        'layout': {},
+        'paint' : {
+          'fill-color'  : polygon.color,
+          'fill-opacity': polygon.opacity,
```

Comment:
Can we avoid using notation like 'id' for objects? As for - 'fill-color'. It doesn't work if we use camelCase('fillColor')?

Reply:
Every polygon needs a unique id to be added on map, I think it is not a good idea to ask users to create a unique name(id) for every polygon. 'fill-color' and 'fill-opacity' doesn't work if we use camelCase.